### DISCUSSION TITLE
Chameleon Tools菜单的显示与隐藏

### DISCUSSION THREAD

**Discussion by reinalin** [cgerchenhp/UE_TAPython_Plugin_Release#12] [General]
Chameleon Tools菜单能不能和SketchTools、MenusOnToolbar一样可以在Config里控制显示与隐藏？

**Comment by cgerchenhp** [comment]
Yes, we can hide the "Chameleon Tool" with `"enabled": false` in file: MenuConfig.json

```JSON
{
    "name": "Minimal Example",
    "ChameleonTools": "../Python/Example/MinimalExample.json",
    "enabled": false,
    "icon": {
        "ImagePathInPlugin": "Resources/flash_32x.png"
    }
},
```

**Reply by reinalin** [reply]
谢谢回复
我还有一个问题，关于widget内容的初始化, 如果用户对某些内容进行了修改，例如：lineedit内容，radiobutton选择，checkbox状态等等，能否将这些内容保存下来，关闭widget再打开依然是用户上一次的设置

**Comment by cgerchenhp** [comment]
可以的。比如下面的示例中，在`OnClosePyCmd`中用你喜欢的格式，将ui状态保存到一个单独的存盘文件中，比如放到工程的Saved目录下。然后在`chameleon_example.load_ui_status()`中，通过`self.data.set_text(...), self.set_combo_box_selected_item(...), self.data.set_is_checked(...)...`等重新设置界面状态。事实上直接在`"OnClosePyCmd"`中修改界面的json文件也是可行的，虽然不推荐。


Q: The question above is whether we can restore the user modified interface the next time we open it. Such as editable text radio button or checkbox.
A: Yes, we can save the status of the widget to a local file, for instance a json/ini file in <Your_project>/Saved/ folder, then load and restore the ui status with the `self.data.set_text(...), self.set_combo_box_selected_item(...), self.data.set_is_checked(...)...`. The event `"OnClosePyCmd"` is used for cleaning up and saving status. 

Or even modify the ui json file directly on `"OnClosePyCmd"`, which is a little hacky.

```JsonPath
{
    "TabLabel": "Minimal Example",
    ...
    "InitPyCmd": "import Example; chameleon_example = Example.MinimalExample.MinimalExample(%JsonPath); chameleon_example.load_ui_status()",
    "OnClosePyCmd": "chameleon_example.save_ui_status()",
    "Root":
    {
        ...
    }
}
```
